plugins/targetlocker/inmemory/inmemory.go

```diff
@@ -95,30 +111,48 @@ func broker(lockRequests, unlockRequests, checkLocksRequests <-chan request, don
 			}
 			req.err <- lockErr
 		case req := <-unlockRequests:
+			if err := validateRequest(req); err != nil {
+				req.err <- fmt.Errorf("unlock request: %w", err)
+			}
 			log.Debugf("Requested to transactionally unlock %d targets: %v", len(req.targets), req.targets)
+			var unlockErr error
 			for _, t := range req.targets {
-				if _, ok := locks[*t]; ok {
-					delete(locks, *t)
+				if l, ok := locks[*t]; ok {
+					if l.owner == req.owner {
+						delete(locks, *t)
+					} else {
+						unlockErr = fmt.Errorf("unlock request: denying unlock request from job ID %d on lock owned by job ID %d", req.owner, l.owner)
+					}
 				} else {
-					log.Debugf("Target is not locked, but received unlock request: %+v", t)
+					// XXX should we roll back the unlock if a target fails?
+					unlockErr = fmt.Errorf("unlock request: target is not locked: %+v", t)
+					break
 				}
 			}
-			req.err <- nil
+			req.err <- unlockErr
```

Comment:
unlock requests always succeeded, but since we now check for lock owners, having an error is necessary